src/app/public/modules/flyout/flyout-instance.ts

```diff
@@ -11,11 +12,16 @@ import {
   SkyFlyoutMessageType
 } from './types';
 
-export class SkyFlyoutInstance<T> {
+export class SkyFlyoutInstance<T> implements OnDestroy {
   public closed = new EventEmitter<void>();
   public componentInstance: T;
   public isOpen = true;
 
+  public iterator = {
```

Comment:
Another (potentially better) option would be to create a separate component for the iterator buttons, instead of a separate class. We're essentially mimicking component behavior by having an input to disable the buttons, and outputs to listen for click events. The flyout component would then communicate with the iterator component.

One potential benefit of this approach is that the iterator component could be later extracted if another component needs it (or if a consumer needs an iterator for something else they're doing).

```
<sky-flyout-iterator
  [nextButtonDisabled]=""
  [previousButtonDisabled]=""
  (nextButtonClick)=""
  (previousButtonClick)=""
>
</sky-flyout-iterator>
```

Reply:
After considering your proposals, I think I'm leaning more toward the iterator component. It gives us more flexibility in the future. I'll work on that...

Reply:
done. flyout-iterator component created